src/components/article-body/body-components/codepanel.tsx

```diff
@@ -1,52 +1,92 @@
+    // This seems to be a CodeMirror issue, but is especially apparent here without line wrapping.
+    // https://github.com/codemirror/codemirror5/issues/5639 describes a similar issue.
+
+    // We're utilizing the Intersection Observer API (https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)
+    // to force the component to rerender when the Code Block element comes into the viewport.
+
+    const [, forceUpdate] = useReducer(x => x + 1, 0); // https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate
+    const wrapper = useRef<HTMLDivElement>(null);
+
+    const onVisible = useCallback(
+        (
+            entries: IntersectionObserverEntry[],
+            observer: IntersectionObserver
+        ) => {
+            // When this gets called with the element intersecting the viewport, we rerender and disconnect the observer.
+            // We are only ever registering 1 element to the observer, so we can disconnnect whenever this gets called with an entry.
+            if (entries.length && entries[0].isIntersecting) {
+                forceUpdate();
+                observer.disconnect();
             }
-        });
+        },
+        []
+    );
+
+    useEffect(() => {
+        let observer: IntersectionObserver;
+        if (wrapper.current) {
+            // Set up the observer to execute the onVisible callback when 1% of the block is in the viewport.
+            observer = new IntersectionObserver(onVisible, {
+                threshold: 0.01,
+            });
+            observer.observe(wrapper.current);
+        }
+
+        return () => {
+            // Clean up and disconnect, needed if the user hasn't scrolled the code block into view.
+            if (observer) {
+                observer.disconnect();
+            }
+        };
+    }, [wrapper]);
+
+    const snippets: Snippet[] = [];
+    children.map((tab: Tab) => {
+        const { type, value } = tab;
+        if (type === 'code' && value) {
+            const { lang } = tab;
+            snippets.push({
+                language: getFloraLanguage(lang ? lang.toLowerCase() : ''),
+                code: value,
+            });
+        }
     });
     return (
         snippets.length > 0 && (
-            <CodePanel
-                snippets={snippets}
-                height={getCodeHeight(maxNumberOfLines)}
-                lineWrapping={false}
-            />
+            <div
+                sx={{
+                    '.CodeMirror-simplescroll': { isolation: 'isolate' },
+                    '.CodeMirror-simplescroll-horizontal div': {
+                        background: '#e7eeec',
+                    },
+                }}
+                ref={wrapper}
+            >
+                <CodePanel
+                    snippets={snippets}
+                    height="auto"
+                    lineWrapping={false}
+                />
```

Comment:
@mcolella14 for some reason the rendering of code tabs is not matching the story book of flora - not sure why - can you please check. 
<img width="895" alt="image" src="https://github.com/mongodb/devcenter/assets/9810710/90ccc679-8516-4378-8f66-7b3c2997660a">
<img width="775" alt="image" src="https://github.com/mongodb/devcenter/assets/9810710/ae7c26dc-6082-42b9-acb6-88b63e4ee41d">